packages/next-query-params/src/NextQueryParamProvider.tsx

```diff
@@ -11,7 +11,7 @@ type Props = Omit<
 
 function NextQueryParamProvider({children, ...rest}: Props) {
   const router = useRouter();
-  const match = router.asPath.match(/[^?]+/);
+  const match = router.asPath.match(/[^?|#]+/);
```

Comment:
Is there a particular reason why you added the `|`? The match tries to capture the pathname, which excludes the query params and hash part. Shouldn't this work without the `|`?

Reply:
Btw. possibly we could move the regex to a variable, so we don't have to duplicate it here and below.

Reply:
The `|` is actually not necessary indeed, good one! I've also moved the regex to a variable :)